git/parse_test.go

```diff
@@ -41,3 +132,48 @@ func TestParseSignature(t *testing.T) {
 
 	}
 }
+
+func TestParseTag(t *testing.T) {
+	t.Log("Test Parse Signed Tag")
+	tagReader := strings.NewReader(fakeSignedTagTxt)
+	tagRC := ioutil.NopCloser(tagReader)
+	fakeGitObject := gitObject{ObjTag, tagReader.Size(), tagRC}
+	tag, err := parseTag(fakeGitObject)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+	}
+	if tag.Message != FakeSignedTag.Message &&
+		tag.GPGSig != FakeSignedTag.GPGSig &&
+		tag.Tag != FakeSignedTag.Tag &&
+		tag.Tagger != FakeSignedTag.Tagger &&
+		tag.Object != FakeSignedTag.Object &&
+		tag.ObjType != FakeSignedTag.ObjType {
+		t.Logf("[E] tag returned:\n%s\n", tag)
+		t.Logf("[E] tag expected:\n%s\n", FakeSignedTag)
+		t.Fail()
```

Comment:
There is `reflect.DeepEqual`, but then `fakeGitObject`'s source field would have to be set to `nil` directly before the use.